src/nfa/nfa-intersection.cc

```diff
@@ -117,12 +117,14 @@ Nfa intersection(const Nfa& lhs, const Nfa& rhs, bool preserve_epsilon, ProductM
         pair_to_process = *pairs_to_process.cbegin();
         pairs_to_process.erase(pair_to_process);
         // Compute classic product for current state pair.
-        Mata::Util::SynchronizedUniverzalIterator<TransSymbolStates> sui(2);
-        sui.push_back(lhs.transitionrelation[pair_to_process.first]);
-        sui.push_back(rhs.transitionrelation[pair_to_process.second]);
-
-        while (sui.advance()) {
-            std::vector<TransitionList::const_iterator> moves = sui.get_current();
+        Mata::Util::SynchronizedUniversalIterator<Mata::Util::OrdVector<TransSymbolStates>::const_iterator> sync_iterator(2);
+        Mata::Util::push_back(sync_iterator,lhs.transitionrelation[pair_to_process.first]);
+        Mata::Util::push_back(sync_iterator,rhs.transitionrelation[pair_to_process.second]);
+        //sync_iterator.push_back(lhs.transitionrelation[pair_to_process.first]);
```

Comment:
I would advise not to leave commented out code in PRs. It will only confuse others and yourself in the future.

Reply:
okok, thanks, done